game/game/compatibility/ikarus.h

```diff
@@ -30,38 +30,39 @@ class Ikarus : public ScriptPlugin {
       ptr32_t stack_stackPtr = 0;            // 76
       };
 
-    void  mem_setupexceptionhandler         (Daedalus::DaedalusVM &vm);
-    void  mem_getaddress_init               (Daedalus::DaedalusVM &vm);
-    void  mem_printstacktrace_implementation(Daedalus::DaedalusVM &vm);
-    void  mem_getfuncptr                    (Daedalus::DaedalusVM &vm);
-    void  mem_replacefunc                   (Daedalus::DaedalusVM &vm);
-    void  mem_searchvobbyname               (Daedalus::DaedalusVM &vm);
+    void  mem_setupexceptionhandler         ();
+    void  mem_getaddress_init               ();
+    void  mem_printstacktrace_implementation();
+    int   mem_getfuncptr                     (int func);
+    void  mem_replacefunc                   (int dest, int func);
+    int   mem_searchvobbyname                (std::string_view name);
 
     // ## Basic Read Write ##
-    void  mem_readint                       (Daedalus::DaedalusVM &vm);
-    void  mem_writeint                      (Daedalus::DaedalusVM &vm);
-    void  mem_copybytes                     (Daedalus::DaedalusVM &vm);
-    void  mem_getcommandline                (Daedalus::DaedalusVM &vm);
+    int         mem_readint       (int address);
+    void        mem_writeint      (int address, int val);
+    void        mem_copybytes     (int src, int dst, int size);
+    std::string mem_getcommandline();
 
     // pointers
-    void  mem_ptrtoinst                     (Daedalus::DaedalusVM &vm);
+    int  mem_ptrtoinst            (int address);
 
     // ## Basic zCParser related functions ##
-    void  _takeref    (Daedalus::DaedalusVM &vm);
-    void  _takeref_s  (Daedalus::DaedalusVM &vm);
-    void  _takeref_f  (Daedalus::DaedalusVM &vm);
+    int  _takeref    (int val);
```

Comment:
You are right, I should probably use `int32_t` instead of just `int`. This isn't critical though since all relevant systems use at least 32-bits for an `int` according to [cppreference](https://en.cppreference.com/w/cpp/language/types) (see `LP32`).

I'm adding it as a ToDo in _phoenix_ and I'll get to it after this PR is merged :)